### ISSUE TITLE
[R-package] How to remove these infos

### ISSUE THREAD

**Issue by nobugforerer** [microsoft/LightGBM#5347]
![1656775005422](https://user-images.githubusercontent.com/92627523/177006617-7721b8f8-7628-44ea-bd15-9751567845f2.png)

**Comment by nobugforerer** [comment]
above happened when i train a model.
i have tried set verbose=-1 but it does not help

**Comment by jameslamb** [comment]
Thanks for using LightGBM!

Here's one example of how the `verbose` parameter to `lgb.train()` can be used to control whether or not evaluation metric messages are printed.

```r
library(lightgbm)

data("agaricus.train", package = "lightgbm")

dtrain <- lgb.Dataset(
  data = agaricus.train$data
  , label = agaricus.train$label
)

bst <- lgb.train(
  data = dtrain
  , nrounds = 10L
  , params = list(
    objective = "binary"
    , num_leaves = 7L
  )
  , valids = list("train" = dtrain)
  , verbose = -1L
)
```

On both `{lightgbm}` v3.3.2 installed from CRAN and on the latest development version as of today (https://github.com/microsoft/LightGBM/commit/25e32e94636b0d81b77c20d26635914ee330266d), this does not produce any log messages.

Changing to `verbose = 1L` in the call to `lgb.train()`, I see the following in the logs.

```text
[1] "[1]:  train's binary_logloss:0.601483"
[1] "[2]:  train's binary_logloss:0.526954"
[1] "[3]:  train's binary_logloss:0.464663"
[1] "[4]:  train's binary_logloss:0.41188"
[1] "[5]:  train's binary_logloss:0.364222"
```

This is just one example. It's difficult for maintainers to give you a specific answer since you've decided not to provide some of the information that was asked for in the issue template, like:

* version of LightGBM you are using
* a reproducible example (specific code you ran)

If this answer doesn't resolve your question, please provide that information and we'll be happy to help.

-----------

One other thing...in the future, please do not post images of code or printed output. Include such things as text instead, so that they can be found by search engines in case other people face the same issue you're facing.

**Comment by jameslamb** [comment]
If you're using `lightgbm()` instead of `lgb.train()`, you can suppress these messages by passing `verbose = -1` to `lightgbm()`.

Consider the following example:

```r
library(lightgbm)

data("agaricus.train", package = "lightgbm")

bst <- lightgbm(
  data = agaricus.train$data
  , label = agaricus.train$label
  , nrounds = 10L
  , params = list(
    objective = "binary"
    , num_leaves = 7L
  )
  , verbose = 1L
)
```

On both v3.3.2 and the latest development version, passing `verbose = -1L` results in training succeeding with 0 log messages produced.

**Comment by nobugforerer** [comment]
As what you said, these logs do not come out by using 'lgb.train', but it does in the case 'lightgbm'

> If you're using `lightgbm()` instead of `lgb.train()`, you can suppress these messages by passing `verbose = -1` to `lightgbm()`.
> 
> Consider the following example:
> 
> ```r
> library(lightgbm)
> 
> data("agaricus.train", package = "lightgbm")
> 
> bst <- lightgbm(
>   data = agaricus.train$data
>   , label = agaricus.train$label
>   , nrounds = 10L
>   , params = list(
>     objective = "binary"
>     , num_leaves = 7L
>   )
>   , verbose = 1L
> )
> ```
> 
> On both v3.3.2 and the latest development version, passing `verbose = -1L` results in training succeeding with 0 log messages produced.

**Comment by nobugforerer** [comment]
sorry, because my complete code needs an excel file to run, it seems like i cannot uploaded it

**Comment by jameslamb** [comment]
> my complete code needs an excel file to run

You could, for example:

* copy my example using `data("agaricus.train", package = "lightgbm")`
* use one of R's built-in datasets ([documentation link](https://stat.ethz.ch/R-manual/R-devel/library/datasets/html/00Index.html))
* provide code that generates sample data with something like `rnorm()`

I've provided a few examples of code that does and does not print these log messages, for v3.3.2 and the latest version of `{lightgbm}`. If you're able to provide a reproducible example, then the difference between my examples and your code might help us narrow down the source of the issue.

Otherwise, maintainers will just have to guess at combinations of inputs that might produce the behavior you're seeing. We cannot commit time to doing that.

**Comment by nobugforerer** [comment]
it works when i set verbose outside the params (before it is passed to params).
but there is a new problem,when i run this:
`data("agaricus.train", package = "lightgbm")

dtrain <- lgb.Dataset(
  data = agaricus.train$data
  , label = agaricus.train$label
)

bst <- lightgbm(
  data = dtrain
  , nrounds = 10L
  , params = list(
    objective = "regression"
    , num_leaves = 7L
      
  )
  ,verbose=-1
  , early_stopping_rounds = 10  

)`

this error happens:

- Error in init(env = env) : 
For early stopping, valids must have at least one element

but when i remove verbose, it performs normal:
`data("agaricus.train", package = "lightgbm")

dtrain <- lgb.Dataset(
  data = agaricus.train$data
  , label = agaricus.train$label
)

bst <- lightgbm(
  data = dtrain
  , nrounds = 10L
  , params = list(
    objective = "regression"
    , num_leaves = 7L
      
  )
  , early_stopping_rounds = 10  

)`

- [LightGBM] [Warning] Auto-choosing row-wise multi-threading, the overhead of testing was 0.001751 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 214
[LightGBM] [Info] Number of data points in the train set: 6513, number of used features: 107
[LightGBM] [Info] Start training from score 0.482113
[1] "[1]:  train's l2:0.20423"
[1] "[2]:  train's l2:0.167388"
[1] "[3]:  train's l2:0.137409"
[1] "[4]:  train's l2:0.113006"
[1] "[5]:  train's l2:0.0920465"
[1] "[6]:  train's l2:0.0750694"
[1] "[7]:  train's l2:0.0613179"
[1] "[8]:  train's l2:0.0501792"
[1] "[9]:  train's l2:0.0411569"
[1] "[10]:  train's l2:0.0338488"


> > my complete code needs an excel file to run
> 
> You could, for example:
> 
> * copy my example using `data("agaricus.train", package = "lightgbm")`
> * use one of R's built-in datasets ([documentation link](https://stat.ethz.ch/R-manual/R-devel/library/datasets/html/00Index.html))
> * provide code that generates sample data with something like `rnorm()`
> 
> I've provided a few examples of code that does and does not print these log messages, for v3.3.2 and the latest version of `{lightgbm}`. If you're able to provide a reproducible example, then the difference between my examples and your code might help us narrow down the source of the issue.
> 
> Otherwise, maintainers will just have to guess at combinations of inputs that might produce the behavior you're seeing. We cannot commit time to doing that.

**Comment by jameslamb** [comment]
Ok thanks very much for that!

The issue here is that as of v3.3.2, the `verbose` parameter in `lightgbm()` was unfortunately doing two jobs:

* determining which (if any) messages to print
* determining whether or not to automatically register the training dataset as a validation dataset

You can see #5203 and #5273 for more details on how that'll change in LightGBM v4.0.0.

To use early stopping on the training data in `lightgbm()` but without any printed messages, use `verbose = -1` and pass the training data through `valids`, like this.

```r
library(lightgbm)

data("agaricus.train", package = "lightgbm")

dtrain <- lgb.Dataset(
  data = agaricus.train$data
  , label = agaricus.train$label
)

bst <- lightgbm(
  data = dtrain
  , nrounds = 10L
  , params = list(
    objective = "regression"
    , num_leaves = 7L
  )
  , verbose = -1L
  , early_stopping_rounds = 10L
  , valids = list(
    "train" = dtrain
  )
)
```

This will work as expected in v3.3.2 and the upcoming v4.0.0 release.

---

I've reformatted your comment above so it's a bit easier to read. Since I think you might be new to GitHub, please see https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code for some details on how to format code in GitHub posts.

**Comment by nobugforerer** [comment]
> Ok thanks very much for that!
> 
> The issue here is that as of v3.3.2, the `verbose` parameter in `lightgbm()` was unfortunately doing two jobs:
> 
> * determining which (if any) messages to print
> * determining whether or not to automatically register the training dataset as a validation dataset
> 
> You can see #5203 and #5273 for more details on how that'll change in LightGBM v4.0.0.
> 
> To use early stopping on the training data in `lightgbm()` but without any printed messages, use `verbose = -1` and pass the training data through `valids`, like this.
> 
> ```r
> library(lightgbm)
> 
> data("agaricus.train", package = "lightgbm")
> 
> dtrain <- lgb.Dataset(
>   data = agaricus.train$data
>   , label = agaricus.train$label
> )
> 
> bst <- lightgbm(
>   data = dtrain
>   , nrounds = 10L
>   , params = list(
>     objective = "regression"
>     , num_leaves = 7L
>   )
>   , verbose = -1L
>   , early_stopping_rounds = 10L
>   , valids = list(
>     "train" = dtrain
>   )
> )
> ```
> 
> This will work as expected in v3.3.2 and the upcoming v4.0.0 release.
> 
> I've reformatted your comment above so it's a bit easier to read. Since I think you might be new to GitHub, please see https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code for some details on how to format code in GitHub posts.

it was a good answer, and thanks for your patience!
